draft-packages/button/KaizenDraft/Button/FilterDrawerButton.tsx

```diff
@@ -0,0 +1,53 @@
+import { Badge } from "@kaizen/draft-badge"
+import classnames from "classnames"
+import * as React from "react"
+import GenericButton, {
+  AdditionalContentProps,
+  GenericProps,
+  LabelProps,
+} from "./components/GenericButton"
+const filterIcon = require("@kaizen/component-library/icons/filter.icon.svg")
+  .default
+const styles = require("./FilterDrawerButton.module.scss")
+
+type FilterButtonProps = {
+  labelText: string
+  numFiltersEnabled: number
+  onClick: (e: any) => void
+  onMouseDown: (e: any) => void
+  reversed?: boolean
+}
+
+export const FilterDrawerButton = ({
+  labelText,
+  numFiltersEnabled,
+  onClick,
+  onMouseDown,
+  reversed,
+}: FilterButtonProps) => (
+  <div
+    className={classnames(styles.buttonEdgeCaseStyling, {
+      [styles.reversed]: reversed,
+    })}
+  >
+    <ButtonAllowingAdditionalContent
+      secondary={true}
+      reversed={reversed}
+      label={labelText}
+      icon={filterIcon}
+      iconPosition="start"
+      additionalContent={renderBadge(numFiltersEnabled)}
+      onClick={onClick}
+      onMouseDown={onMouseDown}
+    />
+  </div>
+)
+
+const ButtonAllowingAdditionalContent = (
+  props: GenericProps & LabelProps & AdditionalContentProps
```

Comment:
This feels nice. Splitting out the `AdditionalContentProps`, `GenericProps` and `LabelProps` gives me a very good indication of the intention + allowable features of this wrapper without needing to dig much deeper in implementation 👍

Reply:
I'm really liking the idea of making GenericProps permissive but have restrictive options for the various button variants. Thanks for the suggestion!